doc/source/f2py/usage.rst

```diff
@@ -327,3 +327,68 @@ resulting package to work, you need to create a file named ``__init__.py``
 (in the same directory as ``add.pyf``). Notice the extension module is
 defined entirely in terms of the ``add.pyf`` and ``add.f`` files. The
 conversion of the .pyf file to a .c file is handled by `numpy.distutils`.
+
+Building with Meson (Examples)
+==============================
+
+Using f2py with Meson
+~~~~~~~~~~~~~~~~~~~~~
+
+Meson is a modern build system recommended for building Python extension
+modules, especially starting with Python 3.12 and NumPy 2.x. Meson provides
+a robust and maintainable way to build Fortran extensions with f2py.
+
+To build a Fortran extension using f2py and Meson, you can use Meson's
+`custom_target` to invoke f2py and generate the extension module. The
+following minimal example demonstrates how to do this:
+
+.. code-block:: meson
+
+   # List your Fortran source files
+   fortran_sources = files('your_module.f90')
+
+   # Find the Python installation
+   py = import('python').find_installation()
+
+   # Create a custom target to build the extension with f2py
+   f2py_wrapper = custom_target(
+     'your_module_wrapper',
+     output: 'your_module.so',
+     input: fortran_sources,
+     command: [
+       py.full_path(), '-m', 'numpy.f2py',
+       '-c', '@INPUT@', '-m', 'your_module'
+     ]
+   )
+
+   # Install the built extension to the Python site-packages directory
+   install_data(f2py_wrapper, install_dir: py.site_packages_dir())
+
+For more details and advanced usage, see the Meson build guide in the
+user documentation or refer to SciPy's Meson build files for real-world
+examples: https://github.com/scipy/scipy/tree/main/meson.build
+
+Building NumPy ufunc Extensions with Meson
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To build a NumPy ufunc extension (C API) using Meson, you can use the
+following template:
+
+.. code-block:: meson
+
+   # List your C source files
+   c_sources = files('your_ufunc_module.c')
+
+   # Find the Python installation
+   py = import('python').find_installation()
+
+   # Create an extension module
+   extension_module = py.extension_module(
+     'your_ufunc_module',
+     c_sources,
+     dependencies: py.dependency(),
```

Comment:
Is this good enough? I would think you need to include numpy headers as well (In my own experiments, which are admittedly quite a while ago, that was non-trivial; see  https://github.com/mhvk/chain_ufunc/blob/main/meson.build).